influxql/ast.go

```diff
@@ -137,6 +135,13 @@ type Statement interface {
 	RequiredPrivileges() ExecutionPrivileges
 }
 
+// HasDefaultDB provides an interface to get the default DB from a Statement.
+type HasDefaultDB interface {
```

Comment:
Could you just add `DefaultDatabase` as an attribute to the `stmt` type, and have it filled in at parse time?